tests/test_diag_table.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+import pytest
+
+from fv3config.config.diag_table import DiagTable, DiagTableField, DiagTableFile
+from fv3config._exceptions import ConfigError
+
```

Comment:
Thanks for the pretty extensive tests! One case I see missing is ensuring that the DiagTable resolves down to an expected string. Currently, a bug where the DiagTable outputs an incorrectly formatted string will not cause the tests to fail, so long as the DiagTable can also load itself from that string. Since ensuring the DiagTable is exactly as required by the Fortran model is an important use case, please add a couple test cases for this.

Reply:
I figured this case was covered by explicitly testing `from_str` and testing the `__repr__` --> `from_str` round trip. But I can add an explicit test for it.

Reply:
I saw that's what you were going for. Unfortunately it's very possible to write code that is lax in how it interprets the input and accepts a wide variety of formatting, and also write a to-string method which was too lax in its interpretation of the standard and has differences in formatting.